security/sshdo/files/macports.patch

```diff
@@ -0,0 +1,38 @@
+--- Makefile.orig	2023-06-19 15:23:20.000000000 +1000
++++ Makefile	2023-06-19 15:58:11.000000000 +1000
+@@ -58,8 +58,8 @@
+ 	@echo "make diff      - Show differences between source and installed versions"
+ 
+ test:
+-	@[ -z "`which python3 2>/dev/null | grep '^/'`" -a -n "`which python 2>/dev/null | grep '^/'`" ] && sed 's/env python3$$/env python/' < ./sshdo > ./sshdo2 && chmod 755 ./sshdo2 && mv ./sshdo2 ./sshdo || true
+-	@[ -z "`which python3 2>/dev/null | grep '^/'`" -a -n "`which python 2>/dev/null | grep '^/'`" ] && sed 's/env python3$$/env python/' < ./test_sshdo > ./test_sshdo2 && chmod 755 ./test_sshdo2 && mv ./test_sshdo2 ./test_sshdo || true
++	@[ -z "`which python@PYTHON_BRANCH@ 2>/dev/null | grep '^/'`" -a -n "`which python 2>/dev/null | grep '^/'`" ] && sed 's/env python3\.11$$/env python/' < ./sshdo > ./sshdo2 && chmod 755 ./sshdo2 && mv ./sshdo2 ./sshdo || true
++	@[ -z "`which python@PYTHON_BRANCH@ 2>/dev/null | grep '^/'`" -a -n "`which python 2>/dev/null | grep '^/'`" ] && sed 's/env python3\.11$$/env python/' < ./test_sshdo > ./test_sshdo2 && chmod 755 ./test_sshdo2 && mv ./test_sshdo2 ./test_sshdo || true
+ 	@./test_sshdo
+ 
+ check: test
+@@ -69,7 +69,7 @@
+ 	if [ ! -f $(DEST_ETCDIR)/$(SSHDOERS) ]; then cp $(SSHDOERS) $(DEST_ETCDIR); chmod 644 $(DEST_ETCDIR)/$(SSHDOERS); fi
+ 	if [ ! -f $(DEST_ETCDIR)/$(SSHDO_BANNER) ]; then cp $(SSHDO_BANNER) $(DEST_ETCDIR); chmod 644 $(DEST_ETCDIR)/$(SSHDO_BANNER); fi
+ 	cp $(BIN) $(DEST_BINDIR); chmod 755 $(DEST_BINDIR)/$(BIN)
+-	[ -z "`which python3 2>/dev/null | grep '^/'`" -a -n "`which python 2>/dev/null | grep '^/'`" ] && sed 's/env python3$$/env python/' < $(BIN) > $(DEST_BINDIR)/$(BIN) || true
++	[ -z "`which python@PYTHON_BRANCH@ 2>/dev/null | grep '^/'`" -a -n "`which python 2>/dev/null | grep '^/'`" ] && sed 's/env python3\.11$$/env python/' < $(BIN) > $(DEST_BINDIR)/$(BIN) || true
+ 	[ -d $(DEST_MANDIR)/man8 ] || mkdir -m 755 $(DEST_MANDIR)/man8
+ 	[ -d $(DEST_MANDIR)/man5 ] || mkdir -m 755 $(DEST_MANDIR)/man5
+ 	cp sshdo.8 $(DEST_MANDIR)/man8; chmod 644 $(DEST_MANDIR)/man8/sshdo.8
+--- sshdo.orig	2023-06-19 15:23:20.000000000 +1000
++++ sshdo	2023-06-19 15:56:00.000000000 +1000
+@@ -1,4 +1,4 @@
+-#!/usr/bin/env python3
++#!/usr/bin/env python@PYTHON_BRANCH@
```

Comment:
```suggestion
+#!@PREFIX@/bin/python@PYTHON_BRANCH@
```

The `reinplace` in the Portfile would have to be expanded to also handle the `@PREFIX@` placeholder, e.g. (untested):

```tcl
post-patch {
     reinplace -W ${worksrcpath} "s|@PYTHON_BRANCH@|${python_branch}|g" Makefile
     reinplace -W ${worksrcpath} "s|@PYTHON_BRANCH@|${python_branch}|g;s|@PREFIX@|${prefix}|g" sshdo test_sshdo
 }
```

Reply:
It's done.